src/main/java/gov/cms/mat/cql_elm_translation/service/DataCriteriaService.java

```diff
@@ -213,4 +213,19 @@ private String buildCriteriaType(String dataType) {
     // e.g for negation: "Assessment, Not Performed" becomes "AssessmentPerformed"
     return dataType.replace(",", "").replace(" ", "").replace("Not", "");
   }
+
+  public List<String> getUsedValuesets(String cql, String accessToken) {
+    CQLTools tools = parseCql(cql, accessToken, cqlLibraryService);
+    return tools.getUsedValuesets();
+  }
+
+  public List<CQLValueSet> getUsedCQLValuesets(String cql, String accessToken) {
+    CQLTools tools = parseCql(cql, accessToken, cqlLibraryService);
+    return tools.getUsedCQLValuesets();
+  }
+
+  public List<CQLCode> getCQLCodes(String cql, String accessToken) {
```

Comment:
Might as well keep with the preceding naming pattern:
```suggestion
  public List<CQLCode> getUsedCQLCodes(String cql, String accessToken) {
```

Reply:
Changed